runtime/gc_glue_java/GlobalCollectorDelegate.cpp

```diff
@@ -97,6 +98,12 @@ MM_GlobalCollectorDelegate::initialize(MM_EnvironmentBase *env, MM_GlobalCollect
 
 	/* Balanced and realtime polices will instantiate their own access barrier */
 	if (_extensions->isStandardGC()) {
+		if (_extensions->isSnapshotAtTheBeginningBarrierEnabled()) {
+			_extensions->sATBBarrierRememberedSet = MM_RememberedSetWorkPackets::newInstance(env, markingScheme->getWorkPackets());
+			if (NULL == _extensions->sATBBarrierRememberedSet) {
+				return false;
+			}
+		}
```

Comment:
hm, I have a feeling this should be in OMR part

Reply:
If you make MM_RememberedSetWorkPackets to take workPackets as a specific MM_WorkPacketsSATB type, it's safe to upcast markingScheme->getWorkPackets() to (MM_WorkPacketsSATB* ), since it's under isSATBEnabled check.

Reply:
I moved this code to OMR.